File: reference-code/bukkit/src/main/java/createcommands/arguments/types/StringsArguments.java
package createcommands.arguments.types;

import dev.jorel.commandapi.CommandAPICommand;
import dev.jorel.commandapi.arguments.EntitySelectorArgument;
import dev.jorel.commandapi.arguments.GreedyStringArgument;
import org.bukkit.entity.Player;

class StringsArguments {
    static {
        // #region greedyStringArgumentsExample
        new CommandAPICommand("message")
            .withArguments(new EntitySelectorArgument.OnePlayer("target"))
            .withArguments(new GreedyStringArgument("message"))
            .executes((sender, args) -> {
                ((Player) args.get("target")).sendMessage((String) args.get("message"));
            })
            .register();
        // #endregion greedyStringArgumentsExample
    }
}

File: reference-code/bukkit/src/main/kotlin/createcommands/arguments/types/StringsArguments.kt
package createcommands.arguments.types

import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.arguments.EntitySelectorArgument
import dev.jorel.commandapi.arguments.GreedyStringArgument
import dev.jorel.commandapi.executors.CommandExecutor
import dev.jorel.commandapi.kotlindsl.anyExecutor
import dev.jorel.commandapi.kotlindsl.commandAPICommand
import dev.jorel.commandapi.kotlindsl.entitySelectorArgumentOnePlayer
import dev.jorel.commandapi.kotlindsl.greedyStringArgument
import org.bukkit.entity.Player

fun stringsArguments() {
    // #region greedyStringArgumentsExample
    CommandAPICommand("message")
        .withArguments(EntitySelectorArgument.OnePlayer("target"))
        .withArguments(GreedyStringArgument("message"))
        .executes(CommandExecutor { _, args ->
            (args["target"] as Player).sendMessage(args["message"] as String)
        })
        .register()
    // #endregion greedyStringArgumentsExample
}

fun stringsArgumentsDSL() {
    // #region greedyStringArgumentsExampleDSL
    commandAPICommand("message") {
        entitySelectorArgumentOnePlayer("target")
        greedyStringArgument("message")
        anyExecutor { _, args ->
            (args["target"] as Player).sendMessage(args["message"] as String)
        }
    }
    // #endregion greedyStringArgumentsExampleDSL
}